typescript.js

```diff
@@ -19,7 +19,11 @@ module.exports = {
             https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.ts
         */
 
-        // None
+        // The following elevate recommended rules from warning to error
+        '@typescript-eslint/explicit-module-boundary-types': 'error',
```

Comment:
[@typescript-eslint/explicit-module-boundary-types](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md)
We already enabled [@typescript-eslint/explicit-function-return-type](https://github.com/ni/javascript-styleguide/blob/2aa42157a52b67ec25615ee9f0f370ea01afa768/typescript.js#L53).
**Proposal**
Assuming we're extending the recommended set (are we?), disable the rule.
`'@typescript-eslint/explicit-module-boundary-types': 'off'`

Reply:
Do we have a rule the enforces explicit types for function arguments or is it just enforced by the TypeScript compiler option `noImplicitAny`? I think @typescript-eslint/explicit-module-boundary-types enforces this, but @typescript-eslint/explicit-function-return-type does only enforce the return type to be explicit.

Reply:
Good catch, I haven't found one. That brings up a few additional points.
- I do think `noImplicitAny` would enforce the parameters equivalently. This rule allows default parameters.
- If we aren't enabling `noImplicitAny`, do we really need to document every module as if it were public, like a library? Most application modules are not public. I'm not against enabling `noImplicitAny`, though. If we do enable it, the less rules the better performance. ESLint is quite slow in testing.

Reply:
Agreed in person to leave this rule enabled as an error.